packages/@sanity/codegen/src/typescript/findQueriesInPath.ts

```diff
@@ -0,0 +1,85 @@
+import fs from 'node:fs/promises'
+import {createRequire} from 'node:module'
+import {join} from 'node:path'
+
+import {type TransformOptions} from '@babel/core'
+import debug from 'debug'
+import glob from 'fast-glob'
+import {mergeMap, Observable} from 'rxjs'
+
+import {type NamedQueryResult} from './expressionResolvers'
+import {findQueriesInSource} from './findQueriesInSource'
+
+const $debug = debug('sanity:codegen:findQueries:debug')
+const $trace = debug('sanity:codegen:findQueries:trace')
+
+const require = createRequire(__filename)
+
+const defaultBabelOptions = {
+  extends: join(__dirname, '..', '..', 'babel.config.json'),
+}
+
+type FindQueriesReturnValue = {
+  filename: string
+  queries: Map<string, NamedQueryResult>
+  error?: Error
+}
+
+// Holds all queries found in the source files
+const allQueries = new Set()
+
+export function findQueriesInPath({
+  path,
+  babelOptions = defaultBabelOptions,
+  resolver = require.resolve,
+}: {
+  path: string | string[]
+  babelOptions?: TransformOptions
+  resolver?: NodeJS.RequireResolve
+}): Observable<FindQueriesReturnValue> {
```

Comment:
Instead of an observable and rxjs, how would you feel about this being an async generator? `globby.globStream` returns a readable so it has the ability to do [`for await`](https://github.com/sindresorhus/globby/blob/b0d7330942a3d36a12056f3a1257f676a534b6ea/readme.md#globbystreampatterns-options) and instead of `subscriber.next`, you'd `yield` the next result and errors would bubble automatically.

Reply:
No strong opinions tbh. I'll give it a spin and see how much work to refactor